### DISCUSSION TITLE
Import reejs as a es6 module via URL

### DISCUSSION THREAD

**Discussion by jikkuatwork** [rovelstars/reejs#31] [Q&A]
Hi can we import `reejs` as an es6 module?

Say:

``` js
import Import from "https://reejs-cdn-url";
console.log(Import)
```
If this is possible, can someone point to me the CDN url for it?

**Comment by renhiyama** [comment]
Hey, could I know the usecase for this? I could be able to help much better if I know the usecases, and which runtime are you going to run this on. Please let me know at the earliest @jikkuatwork

**Reply by jikkuatwork** [reply]
Thanks for replying fast! 

Usecase:
I tend build a bunch of CLI tools and I typically do it in Ruby where if you have the library globally you can simply call `require 'something'` and it works. I love that workflow. 

But it definetly has one downside, the moment I share the script file or if its a new system without the gem, it fails and usually just installing it will solve it.

I was dreaming up about a workflow where in my scripts are in JS and I get URL based imports with transparent downloading by the runtime. Now, immediately my scripts become universally shareable and its so much easier to maintain.

Something like this:

``` js
import Import from "https://reejs-cdn-url";
const FakeColors = Import('https://cdn-to-fake-colors')

console.log(`You are being ${FakeColors('red', 'watched')}!`)
```

If this is possible by wrapping `bun` or any runtime, I would still use it so that my DX is much better. 

Inspiration:
I am working on a workflow now that doesn't require any bundling for frontend and its amazing how much can be done without the need for bundling. And I started using a ton of projects with just es6 imports and found the experience so smooth, a single file to import and my components come to life in any page.

**Reply by renhiyama** [reply]
You are forgetting something. The runtime itself must support URL Imports if you want it to work like that. Unless you make use of fetch APIs and eval/any similar method, you cant use URL Imports to import reejs in order to use reejs' URL Imports.

However this sounds like an cool implementation. I will look into this idea soon and let you know!

**Reply by renhiyama** [reply]
Tldr; You need to run (probably a decent 3 lines or so) code at the start of the code to import reejs' Import from online, and then reejs can make sure to work from there.

**Reply by renhiyama** [reply]
Apparently, the only place where your idea would work is deno. But I believe you dont want to use deno for your usecase?

**Reply by jikkuatwork** [reply]
Deno supports url imports? I thought I already tested it failed!

**Reply by renhiyama** [reply]
Deno supports url imports. It goes as far as to provide both static (on top of your code) url imports, as well as dynamic url imports (using async functions). If you are not bound by what runtime you can use, you can continue using Deno. However if you wish to use Nodejs/Bun, lmk and I will write a script to get reejs working as you wanted.

**Reply by jikkuatwork** [reply]
Yes! Please, I prefer Bun over Deno any day!
